app/graphql/resolvers/add_asset_to_submission_resolver.rb

```diff
@@ -1,25 +1,36 @@
 # frozen_string_literal: true
 
 class AddAssetToSubmissionResolver < BaseResolver
-  def valid?
-    true
-  end
+  include Resolvers::Submissionable
 
   def run
-    submission = Submission.find_by(id: @arguments[:submission_id])
-    unless submission
-      raise(GraphQL::ExecutionError, 'Submission from ID Not Found')
-    end
+    check_submission_presence!
```

Comment:
I extracted lookup by submission id / external uuid to a separate module (Resolvers::Submissionable) to avoid code duplication.